CMS-Enterprise/mint-app
# Task/mint 3162 mto card be

# MINT-3162, [MINT-3191](https://jiraent.cms.gov/browse/MINT-3191), [MINT-3190](https://jiraent.cms.gov/browse/MINT-3190), MINT-3189, [MINT-3188](https://jiraent.cms.gov/browse/MINT-3188)




## Description

1. Add Lockable section
2. Add RecentEdit resolver
3. Add parent level status resolver
4. Add milestone resolver


Note, both 2 and 3 are subject to change based on further implementation (change history and the marked as ready for review status)


## How to test this change

1.  Use Postman to
3. Create a Model Plan
5. Query the status and recent edits using postman
   a. should show ready to start and nil
6. Add a category and query it again. You should see status in progress and date should be the created date of the category
7. Do the same with a milestone
8. Recent edit should update to the milestone date.
9. Date should update upon editing as well.
10. Query Milestones on the MTOMatrix level with postman.

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
